### DISCUSSION TITLE
Not able to connect Metamask or Trust Wallet to the demo project

### DISCUSSION THREAD

**Discussion by luduvigo** [WalletConnect/web-examples#9] [Q&A]
Hey, thank you for having developed these amazing web examples, that are a very good starting point for integrating Wallet Connect.

Unfortunately, when I try to connect with the demo site or localhost I get errors from MetaMask or Trust Wallet. 
Trust Wallet returns an Invalid Session error message.

On the contrary, when I try to connect to Uniswap using WalletConnect it runs smoothly.

Any suggestion on how to proceed?

Thank you

**Comment by bkrem** [comment]
Hi @luduvigo,

The currently available examples in this repository are all using **WalletConnect v2**, which MetaMask and Trust Wallet have not yet integrated to my knowledge.

Some wallets that already support WalletConnect v2 and can be used with the v2 dapp examples in this repository:

- Steakwallet (https://steakwallet.fi/)
- AlphaWallet (https://alphawallet.com/download-alphawallet-app/)
- Our example v2 wallet in this repository (https://react-wallet.walletconnect.com)

For WalletConnect v1, there's:

- https://github.com/WalletConnect/walletconnect-test-wallet
- https://github.com/WalletConnect/walletconnect-example-dapp
